Attend-IN/app/(tabs)/home/index.js

```diff
@@ -26,9 +30,38 @@ function UserClasses () {
     }
   }, [userId])
 
+  useEffect(() => {
+    (async () => {
+      let { status } = await Location.requestForegroundPermissionsAsync()
+      if (status !== 'granted') {
+        setErrorMsg('Permission to access location was denied')
+        Alert.alert(
+          { errorMsg },
+          'This app requires location access. Please enable location services in your device settings.',
+          [{ text: 'OK', onPress: () => console.log('OK Pressed') }]
+        )
+        return
+      }
+
+      const locationSubscription = Location.watchPositionAsync(
+        { accuracy: Location.Accuracy.BestForNavigation, timeInterval: 30000, distanceInterval: 1 },
+        (newLocation) => {
+          setLocation(newLocation)
+        }
+      )
+      console.log(location.coords) // display user coordinates in termnial
```

Comment:
will this work if locationSubscription isn't called till line 55 and setLocation is used with locationSubscription. wouldn't it be null at first?

Reply:
line 55 is to remove the stored locationSubscription after user logs off. it still works since locationSubscription is called on line 47.